README.adoc

```diff
@@ -0,0 +1,247 @@
+
+// Create the Pub/Sub client.
+PubsubClient client = PubsubClient
+        .builder()
+        .setAccessTokenCache(accessTokenCache)
+        .setHttpClient(httpClient)
+        .setMeterRegistry(meterRegistry)
+        .build();
+```
+
+You can create a publisher and publish a message as follows:
+
+```java
+// Create the publisher.
+String projectName = "awesome-project";
+PubsubPublisherConfig publisherConfig = PubsubPublisherConfig
+        .builder()
+        .setProjectName(projectName)
+        .setTopicName("awesome-topic")
+        .build();
+PubsubPublisher publisher = PubsubPublisher
+        .builder()
+        .setClient(client)
+        .setConfig(publisherConfig)
+        .build();
+
+// Publish a message.
+publisher
+        .publish(new PubsubPublishRequest(
+                Collections.singletonList(
+                        new PubsubDraftedMessage(
+                                "Yolo like no is watching you!"
+                                        .getBytes(StandardCharsets.UTF_8)))))
+        .subscribeOn(scheduler)
+        .subscribe();
+```
+
+You can create a subscriber and start receiving messages from a subscription as
+follows:
+
+```java
+// Create the message consumer.
+PubsubSubscriber.PullResponseConsumer pullResponseConsumer =
+        pullResponse -> {
+            int messageCount = pullResponse.getReceivedAckableMessages().size();
+            System.out.format("Just got awesome %d messages!%n", messageCount);
+            List<String> ackIds = pullResponse
+                    .getReceivedAckableMessages()
+                    .stream()
+                    .map(PubsubReceivedAckableMessage::getAckId)
+                    .collect(Collectors.toList());
+            PubsubAckRequest ackRequest = new PubsubAckRequest(ackIds);
+            return Mono.just(ackRequest);
+        };
+
+// Create a subscriber.
+PubsubSubscriberConfig subscriberConfig = PubsubSubscriberConfig
+        .builder()
+        .setProjectName(projectName)
+        .setSubscriptionName("awesome-subscription")
+        .build();
+PubsubSubscriber subscriber = PubsubSubscriber
+        .builder()
+        .setClient(client)
+        .setConfig(subscriberConfig)
+        .setPullResponseConsumer(pullResponseConsumer)
+        .setMeterRegistry(meterRegistry)
+        .setScheduler(scheduler)
+        .build();
+
+// Start the subscriber.
+subscriber
+        .start()
+        .subscribe(consumption -> {
+            int pulledMessageCount = consumption
+                    .getPullResponse()
+                    .getReceivedAckableMessages()
+                    .size();
+            int ackedMessageCount = consumption
+                    .getAckRequest()
+                    .getAckIds()
+                    .size();
+            System.out.format(
+                    "Pulled %d, acked %d!%n",
+                    pulledMessageCount, ackedMessageCount);
+        });
+```
+
+== Design Rationale
```

Comment:
Not related to the tech review, but...

This section looks more like a complain rather then the rationale ;) I would make it more dry, without emotions and personal story, and focus more on the benefits of this library over existing approaches (that are most probably used by 100s of companies already, and "reactive" for them is a fancy word that creates problems, not solves it :) ). Just my 2c

Reply:
I am sad to admit that I see your point and you are right. But I still feel an urge to share the historical perspective to explain people the technical (and emotional!) state the idea is stemming from. I believe this might create a sincere developer-to-developer communication channel to get the message across better -- in the sense that, "Yeah bro, I've been there, done that. I feel your pain." Do you think renaming the title to `F.A.Q.` will address your concern?

Reply:
I would then have both - Design Rationale (dry, emotionless, technical description of what the project is doing) and a section describing why you created it (blameless) 👍